collectors/apps.plugin/apps_plugin.c

```diff
@@ -4134,6 +4137,68 @@ static int check_capabilities() {
 }
 #endif
 
+netdata_mutex_t mutex = NETDATA_MUTEX_INITIALIZER;
+
+void *reader_main(void *arg __maybe_unused) {
+    char buffer[PLUGINSD_LINE_MAX + 1];
+
+    while(fgets(buffer, PLUGINSD_LINE_MAX, stdin)) {
+
+        char *words[PLUGINSD_MAX_WORDS] = { NULL };
+        pluginsd_split_words(buffer, words, PLUGINSD_MAX_WORDS, NULL, NULL, 0);
+
+        if(words[0] && strcmp(words[0], PLUGINSD_KEYWORD_FUNCTION) == 0) {
+            char *transaction = words[1];
+            char *timeout_s = words[2];
+            char *function = words[3];
+
+            if(!transaction || !*transaction || !timeout_s || !*timeout_s || !function || !*function) {
+                error("Received incomplete %s (transaction = '%s', timeout = '%s', function = '%s'). Ignoring it.",
+                      words[0],
+                      transaction?transaction:"(unset)",
+                      timeout_s?timeout_s:"(unset)",
+                      function?function:"(unset)");
+            }
+            else {
```

Comment:
Another coverity for these lines:

```sh
*** CID 357333:  Insecure data handling  (TAINTED_SCALAR)
/collectors/apps.plugin/apps_plugin.c: 4163 in reader_main()
4157                           words[0],
4158                           transaction?transaction:"(unset)",
4159                           timeout_s?timeout_s:"(unset)",
4160                           function?function:"(unset)");
4161                 }
4162                 else {
>>>     CID 357333:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "*timeout_s" to "str2i", which uses it as a loop boundary.
4163                     int timeout = str2i(timeout_s);
4164                     if(timeout <= 0) timeout = PLUGINS_FUNCTIONS_TIMEOUT_DEFAULT;
4165     
4166                     internal_error(true, "FUNCTION '%s' with transaction id '%s' received", function, transaction);
4167     
4168                     netdata_mutex_lock(&mutex);

** CID 357339:  Concurrent data access violations  (MISSING_LOCK)
/libnetdata/popen/popen.c: 82 in netdata_popen_tracking_init()
```

Reply:
I made all str2x functions have a maximum string they parse.